File: algorithms/bubble_sort.cpp
#include "library.h"

void sort(int n, int* arr)
{
    // double tab[n];
    for(int i = 0; i < n-1; i++ ){
        for(int j = n - 1; j > i; j--){
            if(arr[j]<arr[j-1]) replace(arr[j], arr[j-1]);
        }
    }
}

int main()
{
    int arr[] = {9, 5, 7, 3, 1};
    sort(5, arr);
    print(5, arr);
}

File: algorithms/bubble_sort.php
<?php
    include_once "library.php";

    function BSort(& $list)
    {
        for ($i=0; $i < sizeof($list) ; $i++) { 
            for ($j=sizeof($list) - 1; $j > $i ; $j--) { 
                if ($list[$j-1] > $list[$j]) {
                    replace($list[$j-1], $list[$j]);
                }
            }
        }
    }

    $list = array( 3, 2, 6, 5, 1, 4 );
    BSort($list);
    printArray($list);
?>